### PR TITLE
Add Geo Models for Division, District, and Thana

### PR DESCRIPTION
**Pull request by TasfiqulGhani**
### Summary
This PR introduces three new models to the project: `Division`, `District`, and `Thana`. These models are designed to represent geographical divisions with associated polygon data.

### Changes:
- **Division Model**: Added to represent major administrative divisions. Each division includes a name, a polygon for geographical boundaries, and an optional description.
- **District Model**: Added to represent districts within divisions. Each district is associated with a division, has a polygon field for geographical boundaries, and an optional description.
- **Thana Model**: Added to represent thanas within districts. Each thana is linked to a district, includes a polygon for geographical boundaries, and an optional description.

### Admin Panel Enhancements:
- Registered the `Division`, `District`, and `Thana` models in the Django admin panel.
- Configured list views, search fields, and filters for better management of geographical data in the admin interface.

### Additional Notes:
- These models lay the groundwork for handling geographical data in the project. Further enhancements may include integrating GIS functionalities or adding additional fields as required.

### CODE REVIEW

File: backend/incident_manager/models/incident.py [hunk 1]
```diff
@@ -11,16 +14,30 @@ class Incident(models.Model):
...
+    )
     location = models.PointField(
         geography=True, null=True, default=None
     )  # Using GeoDjango for geo fields
     description = models.TextField()
     additional_info = models.JSONField(
         default=dict
     )  # Set default to an empty dictionary
-    status = models.CharField(max_length=50)
+    status = models.CharField(
```
Comment (skfarhad) [inline backend/incident_manager/models/incident.py:31]: Incident model will have two types of status- 
- validation_status: ("pending", "validated", "rejected")
- task_status:  ("opened", "closed", "postponed")
Task status will indicate that if any volunteer is acted on this to rescue or not, after rescue the status will be "closed", if it's not possible to act then it'll be "postponed". kindly add the properties

**Review by skfarhad — CHANGES_REQUESTED** [review]

File: backend/incident_manager/models/address.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    def __str__(self):
+        return self.name
+
+
+class District(models.Model):
+    name = models.CharField(max_length=100, unique=True)
+    division = models.ForeignKey(
+        Division, on_delete=models.CASCADE, related_name="districts"
+    )
+    polygon = models.MultiPolygonField(null=True, blank=True)
```
Comment (skfarhad) [inline backend/incident_manager/models/address.py:18]: Also, add a property called- `center` with type  `PointField`

File: backend/incident_manager/models/address.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+        return self.name
+
+
+class Thana(models.Model):
+    name = models.CharField(max_length=100, unique=True)
+    district = models.ForeignKey(
+        District, on_delete=models.CASCADE, related_name="thanas"
+    )
+    polygon = models.MultiPolygonField(null=True, blank=True)
+    description = models.TextField(null=True, blank=True)
```
Comment (skfarhad) [inline backend/incident_manager/models/address.py:31]: Also, add a property called- center with type PointField

**Review by skfarhad — CHANGES_REQUESTED** [review]

File: backend/incident_manager/models/incident.py (hunk 1 above)
Comment (TasfiqulGhani) [reply backend/incident_manager/models/incident.py]: Done.

File: backend/incident_manager/models/address.py (hunk 2 above)
Comment (TasfiqulGhani) [reply backend/incident_manager/models/address.py:19]: Done.

File: backend/incident_manager/models/address.py (hunk 3 above)
Comment (TasfiqulGhani) [reply backend/incident_manager/models/address.py:34]: Done.

**Review by skfarhad — APPROVED** [review]
Now it's perfect!

**Comment by dg1223** [comment]
@TasfiqulGhani @skfarhad Hi! I am working on #38 to create a database for admin boundaries using the model in [address.py](https://github.com/skfarhad/hazard_reporting_system/blob/f3f3ba4331a723090fbed76a3a5d61864f5c50dc/backend/incident_manager/models/address.py).

I was looking at the database table structure in address.py. I noticed that it is missing Upazila, which is the admin level between District and Thana. I'm not sure if it is intended. We can create relationships even if we skip this level but I wanted to bring this to your attention nonetheless.

If you want, I can add it to the model in address.py and update the foreign keys accordingly. Let me know.

Here are the official admin levels from the latest available [shapefiles](https://github.com/yasserius/bangladesh_geojson_shapefile).

adm = Administration Regions
- adm0 = Entire Bangladesh
- adm1 = 8 Divisions
- adm2 = 64 Districts/Zilas
- adm3 = 544 Upazilas
- adm4 = 5160 Thanas/Unions

**Comment by skfarhad** [comment]
> @TasfiqulGhani @skfarhad Hi! I am working on #38 to create a database for admin boundaries using the model in [address.py](https://github.com/skfarhad/hazard_reporting_system/blob/f3f3ba4331a723090fbed76a3a5d61864f5c50dc/backend/incident_manager/models/address.py).
> 
> I was looking at the database table structure in address.py. I noticed that it is missing Upazila, which is the admin level between District and Thana. I'm not sure if it is intended. We can create relationships even if we skip this level but I wanted to bring this to your attention nonetheless.
> 
> If you want, I can add it to the model in address.py and update the foreign keys accordingly. Let me know.
> 
> Here are the official admin levels from the latest available [shapefiles](https://github.com/yasserius/bangladesh_geojson_shapefile).
> 
> adm = Administration Regions
> 
> * adm0 = Entire Bangladesh
> * adm1 = 8 Divisions
> * adm2 = 64 Districts/Zilas
> * adm3 = 544 Upazilas
> * adm4 = 5160 Thanas/Unions

@dg1223 you may add a new column `Upazila`.  And, is Union will be one level higher that Thana? Or they are in the same admin level?

**Comment by dg1223** [comment]
> > @TasfiqulGhani @skfarhad Hi! I am working on #38 to create a database for admin boundaries using the model in [address.py](https://github.com/skfarhad/hazard_reporting_system/blob/f3f3ba4331a723090fbed76a3a5d61864f5c50dc/backend/incident_manager/models/address.py).
> > I was looking at the database table structure in address.py. I noticed that it is missing Upazila, which is the admin level between District and Thana. I'm not sure if it is intended. We can create relationships even if we skip this level but I wanted to bring this to your attention nonetheless.
> > If you want, I can add it to the model in address.py and update the foreign keys accordingly. Let me know.
> > Here are the official admin levels from the latest available [shapefiles](https://github.com/yasserius/bangladesh_geojson_shapefile).
> > adm = Administration Regions
> > 
> > * adm0 = Entire Bangladesh
> > * adm1 = 8 Divisions
> > * adm2 = 64 Districts/Zilas
> > * adm3 = 544 Upazilas
> > * adm4 = 5160 Thanas/Unions
> 
> @dg1223 you may add a new column `Upazila`. And, is Union goin to be one level higher than corresponding Thana? Or they are in the same admin level?

@skfarhad Union and Thana are at the same level.

**Pull request merged**